src/model.cpp

```diff
@@ -127,9 +140,87 @@ std::string Model::doSerialisation(FORMATS format) const
+                    for (size_t k = 0; k < variable->equivalentVariableCount(); ++k) {
+                        VariablePtr equivalentVariable = variable->getEquivalentVariable(k);
+                        VariablePair variablePair = std::make_pair(variable, equivalentVariable);
+                        VariablePair reciprocalVariablePair = std::make_pair(equivalentVariable, variable);
+                        bool pairFound = false;
+                        for (VariableMapIterator iter = variableMap.begin(); iter < variableMap.end(); ++iter) {
+                            if (*iter == variablePair || *iter == reciprocalVariablePair) {
+                                pairFound = true;
+                                break;
+                            }
+                        }
+                        if (!pairFound) {
+                            // New unique variable equivalence pair found
+                            variableMap.push_back(variablePair);
+                            // Also create a component map pair corresponding with the variable map pair.
+                            Component* component1 = static_cast<Component*>(variable->getParent());
+                            Component* component2 = static_cast<Component*>(equivalentVariable->getParent());
+                            ComponentPair componentPair = std::make_pair(component1, component2);
+                            componentMap.push_back(componentPair);
+                        }
+                    }
+                }
+            }
+        }
+        // Serialise connections of the model.
+        int componentMapIndex1 = 0;
+        for (ComponentMapIterator iterPair = componentMap.begin(); iterPair < componentMap.end(); ++iterPair) {
+            Component* currentComponent1 = iterPair->first;
+            Component* currentComponent2 = iterPair->second;
+            ComponentPair currentComponentPair = std::make_pair(currentComponent1, currentComponent2);
+            ComponentPair reciprocalCurrentComponentPair = std::make_pair(currentComponent2, currentComponent1);
+            // Check whether this set of map_components has already been serialised
+            bool pairFound = false;
+            for (ComponentMapIterator serialisedIterPair = serialisedComponentMap.begin(); serialisedIterPair < serialisedComponentMap.end(); ++serialisedIterPair) {
+                if (*serialisedIterPair == currentComponentPair || *serialisedIterPair == reciprocalCurrentComponentPair) {
+                    pairFound = true;
+                    break;
+                }
+            }
+            // Continue to the next component pair if the current pair has already been serialised
+            if (pairFound) {
+                ++componentMapIndex1;
+                continue;
+            }
+            // Serialise out the new connection
+            std::string connection = "<connection><map_components";
+            if (currentComponent1 != nullptr) {
```

Comment:
As for the ```> 0``` above, I would drop the ```!= nullptr```, but again that's just me. Similarly, I wouldn't use curly brackets here since the IF statement is a single-line statement, but then again the Google C++ style guide allows it...